api/server_test.go

```diff
@@ -77,3 +123,86 @@ func (s *S) TestIsNotAdmin(c *check.C) {
 	m.ServeHTTP(rec, req)
 	c.Assert("POST", check.Equals, rec.Body.String())
 }
+
+func (s *S) TestRunHTTPServer(c *check.C) {
+	port, err := selectAvailablePort()
+	c.Assert(err, check.IsNil)
+	config.Set("listen", "0.0.0.0:"+port)
+	config.Set("queue:mongo-url", "127.0.0.1:27017")
+	config.Set("queue:mongo-database", "queuedb")
+	RegisterHandler("/foo", "GET", AuthorizationRequiredHandler(authorizedTsuruHandler))
+	defer resetHandlers()
+
+	go RunServer(false)
```

Comment:
Calling RunServer with `false` for the `dryMode` is quite problematic, this is the cause of all the race conditions in the tests and this happens because the test isn't closing the old server and calling shutdown on multiple initialized components. Calling `shutdown.Do()` before the test exits should be able to fix this. However, I'm not sure if some of our packages can safely handle being reinitialized after a shutdown.